pkg/runner/writer_json.go

```diff
@@ -0,0 +1,35 @@
+package runner
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/linuxsuren/api-testing/pkg/apispec"
+	"io"
+)
+
+type jsonResultWriter struct {
+	writer       io.Writer
+	apiConverage apispec.APIConverage
+}
+
+// NewJSONResultWriter creates a new jsonResultWriter
+func NewJSONResultWriter(writer io.Writer) ReportResultWriter {
+	return &jsonResultWriter{writer: writer}
+}
+
+// Output writes the JSON base report to target writer
+func (w *jsonResultWriter) Output(result []ReportResult) (err error) {
+	jsonData, err := json.Marshal(result)
+	if err != nil {
+		return err
+	}
+	fmt.Fprint(w.writer, string(jsonData))
```

Comment:
It's not a good practice to ignore the error. See also the signature of this function.

`func Fprint(w io.Writer, a ...any) (n int, err error) {`